app/models/friendship.rb

```diff
@@ -1,14 +1,14 @@
 class Friendship < ApplicationRecord
-  after_create :create_inverse_relationship
-  after_destroy :destroy_inverse_relationship
-
   belongs_to :user
   belongs_to :friend, class_name: 'User'
 
   validates :user, presence: true
   validates :friend, presence: true, uniqueness: { scope: :user }
   validate :not_self
 
+  after_create :create_inverse_relationship
+  after_destroy :destroy_inverse_relationship
+
```

Comment:
why you are not using self association for friends ?

Reply:
I dont understand how to implement a self association, we can discuss in person about this

Reply:
sure lets discuss